chingu-voyages/v51-tier3-team-33
# Gary/feature/add receipt

# Description
This commit implements the upload receipt functionality for created expenses to add to the database. Now when an expense is created the image of the receipt uploaded at the time of creation will be stored in the database along with the rest of the information about the related expense.

## Related to the following Jira/Trello issue 
As a user, I want to be able to upload a photo of the receipt for an expense, so that I can easily keep track of my expenses and split them with others. - https://radhika-gandi.atlassian.net/browse/CT3PB-48

## Main changes explained:
- Modified addExpense to handle receipt upload
- Delete the Receipt Model, it was no longer necessary.
- Modified Expenses model, to rename receipt_id to receipt_url, typing was changed from ID to String.

## How to test: 
0. Easiest way to test this change is to use Postman or something similar.
0.5. Please verify that you have AWS keys in your env.local file. You can find them in the developer's discussion chat in our discord if you need them, or just ask me.
1. check into current branch.
2. do `pnpm install` and `pnpm run dev` to run this PR locally.
3. Create a post request on Postman to add an expense for an existing group within the database, you can find this on Mongodb or MongoDB compass. The url for the request should be http://localhost:3000/api/groups/*insert groupid here*/addExpense.
4. The body of the request should be entered with form data, Which should be done in key-value pairs. ex: Name: 'Group 1", Description: "This is Group 1", Amount: "4000.00", etc. All key-value pairs should be in "text" format EXCEPT for the "file" (key, which should be in "file" format. This is where you will upload a test file to associate with the expense being created.
5. Verify the successful creation of expense with the receipt url by checking to see if the newly created expense is returned with the reciept_url, click the link to make sure that the image you uploaded can now be seen on AWS. 


## Note:
Modifications were made to the way the api takes the body of the request, it is now handled as form data. This was done to easily implement the "file" uploaded by the User.
